src/pages/ServiceGraph/SummaryPanelGraph.tsx

```diff
@@ -14,46 +29,147 @@ export default class SummaryPanelGraph extends React.Component<SummaryPanelPropT
-          <p>
-            <strong>Labels:</strong>
-            <br />
-            {this.renderLabels()}
-          </p>
+        <div className="panel-heading">Namespace: {servicesLink}</div>
+        <div className="panel-body" hidden={this.state.initialized}>
+          <h3>Click graph background to see summary information...</h3>
+        </div>
+        <div hidden={!this.state.initialized}>
+          <div className="panel-body">
+            <p>{this.renderLabels(this.state.numNodes.toString(), this.state.numEdges.toString())}</p>
+          </div>
           <hr />
-          <div>
-            {this.renderIncomingRpsChart()}
-            {this.renderOutgoingRpsChart()}
-            <ErrorRatePieChart percentError={10} />
+          <RateTable
+            title="Traffic (requests per second):"
+            rate={this.state.rate}
+            rate3xx={this.state.rate3xx}
+            rate4xx={this.state.rate4xx}
+            rate5xx={this.state.rate5xx}
+          />
+          <div style={{ fontSize: '1.2em' }}>
+            <hr />
+            {this.renderRpsChart()}
           </div>
         </div>
       </div>
     );
   }
 
-  renderLabels = () => (
+  private initState = cy => {
+    let numNodes = 0;
+    let numEdges = 0;
+    let rate = 0;
+    let rate3xx = 0;
+    let rate4xx = 0;
+    let rate5xx = 0;
+    let safeRate = (s: string) => {
+      return s === undefined ? 0.0 : parseFloat(s);
+    };
+
+    numNodes = cy.nodes().size();
+    numEdges = cy.edges().size();
+    cy.edges().forEach(edge => {
+      rate += +safeRate(edge.data('rate'));
+      rate3xx += +safeRate(edge.data('rate3XX'));
+      rate4xx += +safeRate(edge.data('rate4XX'));
+      rate5xx += +safeRate(edge.data('rate5XX'));
+    });
+
+    this.setState({
+      initialized: true,
+      numNodes: numNodes,
+      numEdges: numEdges,
+      rate: rate,
+      rate3xx: rate3xx,
+      rate4xx: rate4xx,
+      rate5xx: rate5xx
+    });
+  };
+
+  private renderLabels = (numNodes: string, numEdges: string) => (
     <>
-      <ServiceInfoBadge scale={0.8} style="plastic" leftText="app" rightText="bookinfo" color="green" />
-      <ServiceInfoBadge scale={0.8} style="plastic" leftText="app" rightText="product" color="green" />
-      <ServiceInfoBadge scale={0.8} style="plastic" leftText="version" rightText="v5" color="navy" />
+      <ServiceInfoBadge scale={0.8} style="plastic" leftText="services" rightText={numNodes} color="green" />
+      <ServiceInfoBadge scale={0.8} style="plastic" leftText="edges" rightText={numEdges} color="green" />
     </>
   );
 
-  renderIncomingRpsChart = () => {
-    const dataRps: any = [['x', 1500, 3500, 5500, 7500, 9500, 10500], ['RPS', 350, 400, 150, 850, 50, 220]];
-    const dataErrors: any = [['x', 1500, 3500, 5500, 7500, 9500, 10500], ['Error', 140, 100, 50, 700, 10, 110]];
+  private renderRpsChart = () => {
+    if (this.state.loading) {
+      return <h3>loading chart...</h3>;
```

Comment:
use \<strong> here - keep font sizes consistently the same size across all panels.

Reply:
If this is used in more than one place then it needs to be its own component so the style changes can be managed in one place.

Reply:
Ideally, we would have components all the way down the tree until we get to leaf nodes that would consist of html markup.